examples/firewall/firewall.c

```diff
@@ -0,0 +1,332 @@
+                }
+        }
+
+        if (!dst_flag) {
+            RTE_LOG(INFO, APP, "Firewall NF requires a destination NF with the -d flag.\n");
+            return -1;
+        }
+        if (!debug) {
+            RTE_LOG(INFO, APP, "Running normal mode, use -b flag to enable debug mode\n");
+        }
+        if (!rules_init) {
+            RTE_LOG(INFO, APP, "Please specify a rules JSON file with -f FILE_NAME\n");
+            return -1;
+        }
+        return optind;
+}
+
+
+static int
+packet_handler(struct rte_mbuf* pkt, struct onvm_pkt_meta* meta, __attribute__((unused)) struct onvm_nf_info* nf_info) {
+        struct ipv4_hdr* ipv4_hdr;
+        uint32_t rule = 0;
+        uint32_t track_ip = 0;
+
+        if(onvm_pkt_is_ipv4(pkt)){
+                ipv4_hdr = onvm_pkt_ipv4_hdr(pkt);
+
+                int no_result = rte_lpm_lookup(lpm_tbl, ipv4_hdr->src_addr, &rule);
+                if(!no_result){
+                        switch(rule){
+                        case ONVM_NF_ACTION_TONF:
+                                meta->action = ONVM_NF_ACTION_TONF;
+                                meta->destination = destination;
+                                if (debug) {
+                                    RTE_LOG(INFO, APP, "Packet from source IP %d has been accepted.\n",
+                                            ipv4_hdr->src_addr);
+                                }
+                                break;
+                        default:
+                                // if we can't understand the rule, drop it
+                                meta->action = ONVM_NF_ACTION_DROP;
+                                if (debug) {
+                                    RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n",
+                                            ipv4_hdr->src_addr);
+                                }
+                                break;
+                        }
+                }
+                else {
+                        // no matching rule
+                        // default action is to drop packets
+                        meta->action = ONVM_NF_ACTION_DROP;
+                        if (debug) {
+                            RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n", ipv4_hdr->src_addr);
+                        }
+                }
+        }
+        else {
+                // drop all packets that aren't ipv4
+                if (debug) {
+                    RTE_LOG(INFO, APP, "Packet received not ipv4\n");
+                }
+                meta->action = ONVM_NF_ACTION_DROP;
+        }
+        return 0;
+}
+
+static int lpm_setup(struct onvm_fw_rule** rules, int num_rules) {
+        int i, status;
+
+        req = (struct lpm_request*)rte_malloc(NULL, sizeof(struct lpm_request), 0);
+
+        if(!req) return 0;
+
+        req->max_num_rules = 1024;
+        req->num_tbl8s = 24;
+        req->socket_id = rte_socket_id();
+        req->name[0] = 'f';
```

Comment:
try making the name unique, see https://github.com/sdnfv/openNetVM/blob/develop/onvm/onvm_nflib/onvm_flow_table.c#L78